docs/05-plugin-development.md

```diff
@@ -134,12 +136,43 @@ In case a dynamic import is not passed a string as argument, this hook gets acce
 Next Hook: [`load`](guide/en/#load) if the resolved id that has not yet been loaded, otherwise [`buildEnd`](guide/en/#buildend).
 
-Defines a custom resolver. A resolver can be useful for e.g. locating third-party dependencies. Returning `null` defers to other `resolveId` functions and eventually the default resolution behavior; returning `false` signals that `source` should be treated as an external module and not included in the bundle. If this happens for a relative import, the id will be renormalized the same way as when the `external` option is used.
+Defines a custom resolver. A resolver can be useful for e.g. locating third-party dependencies. Here `source` is the importee exactly as it is written in the import statement, i.e. for
+
+```js
+import { foo } from '../bar,js';
+```
+
+the source will be `"../bar.js""`. The `importer` is the full id of the importing module. When resolving entry points, importer will be `undefined`. You can use this for instance as a mechanism to define custom proxy modules for entry points. The following plugin will only expose the default export from entry points while still keeping named exports available for internal usage:
+
+```js
+async resolveId(source,importer) {
+  if (!importer) {
+    // We need to skip this plugin to avoid an infinite loop
+    const resolution = await this.resolve(source, undefined, { skipSelf: true });
+    // If it cannot be resolved, return `null` so that Rollup displays an error
+    if (!resolution) return null;
+    return `${resolution.id}?entry-proxy`;
+  }
+  return null;
+},
+load(id) {
+  if (id.endsWith('?entry-proxy')) {
+    const importee = id.slice(0, -'?entry-proxy'.length);
+    // Note that this will throw if there is no default export
+    return `export {default} from '${importee}';`;
+  }
+  return null;
+}
+```
+
+When triggering this hook from a plugin via [`this.resolve(source, importer, options)`](guide/en/#thisresolvesource-string-importer-string-options-skipself-boolean--promiseid-string-external-boolean--null), it is possible to pass a custom options object to this hook. While this object will be passed unmodified, plugins should follow the convention of adding a `custom` property with an object where the keys correspond to the names of the plugins that the options are intended for. For details see **TODO: Section about custom parameter passing**.
```

Comment:
Should we show an example of how this `custom` property would look like?

Reply:
I would place this section more to the bottom, so that first the hook return values are explained.

Reply:
> Should we show an example of how this custom property would look like

With the proper link here, maybe the example is not needed?